### PR TITLE
add unicore recipe

### PR DESCRIPTION
**Pull request by endixk**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/unicore/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -x -e
+
+SOURCE_BIN="${PREFIX}/share/${PKG_NAME}-${PKG_VERSION}-${PKG_BUILDNUM}/bin/unicore"
+TARGET_BIN="${PREFIX}/bin/unicore"
+
+pip install torch transformers sentencepiece protobuf
```
Comment (martin-g) [inline recipes/unicore/build.sh:8]:
```suggestion
pip install torch transformers sentencepiece protobuf
cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
```

File: recipes/unicore/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
```
Comment (martin-g) [inline recipes/unicore/build.sh:8]: Installing dependencies with `pip` is not good!
Please use:
- https://anaconda.org/conda-forge/pytorch
- https://anaconda.org/conda-forge/transformers
- https://anaconda.org/conda-forge/sentencepiece
- https://anaconda.org/conda-forge/protobuf
as `host` (and `run` ?!) dependencies

Comment (endixk) [reply recipes/unicore/build.sh:8]: Makes sense, thanks for the recommendation :) We will check our code's integrity with these packages and push another commit.

**Comment by endixk** [comment]
@BiocondaBot please add label

**Comment by endixk** [comment]
@BiocondaBot please fetch artifacts

File: recipes/unicore/build.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+TARGET_CFG="${PREFIX}/etc/path.cfg"
+SOURCE_PY="${PWD}/src/py/predict_3Di_encoderOnly.py"
+TARGET_PY="${PREFIX}/etc/predict_3Di_encoderOnly.py"
+
+mkdir -p "${PREFIX}/etc"
+cp ${SOURCE_CFG} ${TARGET_CFG}
+cp ${SOURCE_PY} ${TARGET_PY}
+
+case $(uname) in
+	Darwin) rustup default stable && rustup target add x86_64-apple-darwin ;;
```
Comment (martin-g) [inline recipes/unicore/build.sh:16]: Why is this needed ?
'compiler("rust")' doesn't do it ?!

File: recipes/unicore/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    - fasttree
+    - raxml
+
+test:
+  commands:
+    - unicore help
+
+about:
+  home: https://github.com/steineggerlab/unicore
+  license: GNU General Public v3
```
Comment (martin-g) [inline recipes/unicore/meta.yaml:45]:
```suggestion
  license: GNU-3.0-or-later
```

File: recipes/unicore/build.sh (hunk 3 above)
Comment (endixk) [reply recipes/unicore/build.sh:16]: I encountered this error from the [previous check](https://github.com/bioconda/bioconda-recipes/actions/runs/12424337180/job/34689755774):
```
02:50:17 BIOCONDA INFO (ERR) error[E0463]: can't find crate for `std`
02:50:17 BIOCONDA INFO (ERR)   |
02:50:17 BIOCONDA INFO (ERR)   = note: the `x86_64-apple-darwin` target may not be installed
02:50:17 BIOCONDA INFO (ERR)   = help: consider downloading the target with `rustup target add x86_64-apple-darwin`
02:50:17 BIOCONDA INFO (ERR) 
02:50:17 BIOCONDA INFO (ERR) error[E0463]: can't find crate for `core`
02:50:17 BIOCONDA INFO (ERR)   |
02:50:17 BIOCONDA INFO (ERR)   = note: the `x86_64-apple-darwin` target may not be installed
02:50:17 BIOCONDA INFO (ERR)   = help: consider downloading the target with `rustup target add x86_64-apple-darwin`
02:50:17 BIOCONDA INFO (ERR) 
02:50:17 BIOCONDA INFO (ERR) For more information about this error, try `rustc --explain E0463`.
02:50:17 BIOCONDA INFO (ERR) error: could not compile `proc-macro2` (build script) due to 1 previous error
```

Comment (endixk) [reply recipes/unicore/build.sh]: let me [check](https://github.com/bioconda/bioconda-recipes/pull/52879/commits/9cd96c81d17c521b2bcd63b179707246c0c08357) whether it was temporary

Comment (endixk) [reply recipes/unicore/build.sh]: Guess it was temporary, weird 🤷‍♂️
It should now ready to be merged I suppose, please check :)

File: recipes/unicore/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('unicore', max_pin="x") }}
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
+    # - {{ compiler('c') }} add these C/C++ compilers if needed
+    # - {{ compiler('cxx') }}
```
Comment (mbhall88) [inline recipes/unicore/meta.yaml:23]:
```suggestion
```

File: recipes/unicore/build.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+TARGET_PY="${PREFIX}/etc/predict_3Di_encoderOnly.py"
+
+mkdir -p "${PREFIX}/etc"
+cp ${SOURCE_CFG} ${TARGET_CFG}
+cp ${SOURCE_PY} ${TARGET_PY}
+
+# case $(uname) in
+# 	Darwin) rustup default stable && rustup target add x86_64-apple-darwin ;;
+# esac
+
```
Comment (mbhall88) [inline recipes/unicore/build.sh:18]:
```suggestion

```

File: recipes/unicore/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  host:
+    - openssl
+  run:
+    # - pytorch-gpu
```
Comment (mbhall88) [inline recipes/unicore/meta.yaml:27]:
```suggestion
```

File: recipes/unicore/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  commands:
+    - unicore help
+
+about:
+  home: https://github.com/steineggerlab/unicore
+  license: GNU-3.0-or-later
+  license_family: GPL3
+  license_file:
+    - LICENSE         # the license file for your package
+    - THIRDPARTY.yml  # this file is generated by cargo-bundle-licenses
```
Comment (mbhall88) [inline recipes/unicore/meta.yaml:49]:
```suggestion
    - LICENSE
    - THIRDPARTY.yml
```

**Review by mbhall88 — APPROVED** [review]
All good once you remove these comments :)
